addon/-private/resources.js

```diff
@@ -4,19 +4,19 @@ import {
 } from '@ember/helper';
 import { createCache, getValue } from '@glimmer/tracking/primitives/cache';
 import { setOwner } from '@ember/application';
-import { destroy, registerDestructor, associateDestroyableChild } from '@ember/destroyable';
+import { destroy, associateDestroyableChild } from '@ember/destroyable';
 
 export class Resource {
-  constructor(ownerOrThunk, args) {
-    if (typeof ownerOrThunk === 'function') {
-      return { definition: this.constructor, args: ownerOrThunk };
-    }
+  static from(thunk) {
```

Comment:
I need to move to a different way of creating Resources because we can't both use the constructor in sub-classes and use the return value for construction in the use decorator (args aren't yet defined when in the use decorator, but the sub-class constructor runs to completion)